### ISSUE TITLE
Failed to build sentencepiece with Windows

### ISSUE THREAD

**Issue by kanchanbarwade** [LibreTranslate/LibreTranslate#463]
Failed to build sentencepiece
ERROR: Could not build wheels for sentencepiece, which is required to install pyproject.toml-based projects

**Comment by NVerwey** [comment]
Having the same issue when running 
`pip install libretranslate` on windows. 

```
error: subprocess-exited-with-error

  × Building wheel for sentencepiece (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [35 lines of output]
      C:\Users\xxx\AppData\Local\Temp\pip-build-env-9e8egaf1\overlay\Lib\site-packages\setuptools\dist.py:745: SetuptoolsDeprecationWarning: Invalid dash-separated options
      !!

              ********************************************************************************
              Usage of dash-separated 'description-file' will not be supported in future
              versions. Please use the underscore name 'description_file' instead.

              By 2023-Sep-26, you need to update your project and remove deprecated calls
              or your builds will no longer be supported.

              See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
              ********************************************************************************
```

```
note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for sentencepiece
Successfully built flask-swagger LTpycld2
Failed to build sentencepiece
ERROR: Could not build wheels for sentencepiece, which is required to install pyproject.toml-based projects
```

**Comment by pierotofy** [comment]
I currently don't have a Windows machine to check on this, but if somebody figures out what the issue is a PR would be most welcome 🙏

**Comment by NVerwey** [comment]
Any updates here? im still getting this error. 

Any hint or workaround would be appreciated!  
Is there maybe any option to install an older(working) version of LibreTranslate on windows?

**Comment by kanchanbarwade** [comment]
Yes, it is working when I installed an older version of Python on windows.
And I did not find any solution to the current version of Python.

**Comment by NVerwey** [comment]
> Yes, it is working when I installed an older version of Python on windows. And I did not find any solution to the current version of Python.

Thanks. I tried it with Python Version 3.9.4 and it worked. 

It appears the Problems are due to changes in Python 3.11.0

**Comment by xKliment** [comment]
> > Yes, it is working when I installed an older version of Python on windows. And I did not find any solution to the current version of Python.
> 
> Thanks. I tried it with Python Version 3.9.4 and it worked.
> 
> It appears the Problems are due to changes in Python 3.11.0

Can confirm its an issue with Python 3.11.X - Version 3.9.0 also Works just fine

**Comment by pierotofy** [comment]
Does this issue still happen? I was able to install via `pip install libretranslate` on Windows 10, Python 3.11.5.

**Comment by matthewfincher** [comment]
I just successfully installed via `pip install libretranslate` on Windows 10, Python 3.11.5

**Comment by LukasThyWalls** [comment]
To update:

Just installed with Python 3.11.6 in Windows 10.
However, It didn't work for me with Python 3.12.2 first for the OP reason (Failed to build sentencepiece).

Thanks.

**Comment by cangSDARM** [comment]
I can install correctly a older version. but i dont know what it is version, libretranslate dont have such command
